be/src/storage/rowset/column_reader.cpp

```diff
@@ -518,12 +518,20 @@ Status ColumnReader::_zone_map_filter(const std::vector<const ColumnPredicate*>&
                                       const ColumnPredicate* del_predicate,
                                       std::unordered_set<uint32_t>* del_partial_filtered_pages,
                                       std::vector<uint32_t>* pages) {
+    LogicalType lt;
+    if (!predicates.empty()) {
+        lt = predicates[0]->type_info()->type();
+    } else if (del_predicate) {
+        lt = del_predicate->type_info()->type();
+    } else {
+        return Status::OK();
+    }
```

Comment:
The type of the predicate may be different from the data type in the segment file, so here it is changed to use the type of the predicate to parse the zone map string